hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/CreateTopLevelDagVisitor.java

```diff
@@ -431,7 +432,11 @@ public Vertex onSlidingWindowAggregate(SlidingWindowAggregatePhysicalRel rel) {
                             windowPolicy,
                             0,
                             aggregateOperation,
-                            resultMapping));
+                            resultMapping,
+                            // Note : resulting project may not have any watermarked field
+                            watermarkedFieldsKeys.isEmpty()
+                                    ? (byte) 0
```

Comment:
I do not understand the comment and the `0` case. This argument is both input and output watermark. Even if there is no watermark in output (eg. lack of `window_end` in projection) there still has to be a watermark on the input. If you use `0` it can be wrong wmKey for given branch.

There is no test for such case.

Reply:
Hm, let me think.